patches/server/0325-Prevent-consuming-the-wrong-itemstack.patch

```diff
@@ -28,18 +28,18 @@ index cb53fc775df387e8bbb099517793daea5ea15e4b..ba4e0bac019b68da166b277028015999
                  this.releaseUsingItem();
              } else {
                  if (!this.useItem.isEmpty() && this.isUsingItem()) {
-+                this.startUsingItem(this.getUsedItemHand(), true); // Paper
++                    this.startUsingItem(this.getUsedItemHand(), true); // Paper
                      this.triggerItemUseEffects(this.useItem, 16);
                      // CraftBukkit start - fire PlayerItemConsumeEvent
                      ItemStack itemstack;
 @@ -3766,8 +3772,8 @@ public abstract class LivingEntity extends Entity {
                      }
  
                      this.stopUsingItem();
--                // Paper start - if the replacement is anything but the default, update the client inventory
--                if (this instanceof ServerPlayer && !com.google.common.base.Objects.equal(defaultReplacement, itemstack)) {
-+                // Paper start
-+                if (this instanceof ServerPlayer) {
-                     ((ServerPlayer) this).getBukkitEntity().updateInventory();
-                 }
-                 // Paper end
+-                    // Paper start - if the replacement is anything but the default, update the client inventory
+-                    if (this instanceof ServerPlayer && !com.google.common.base.Objects.equal(defaultReplacement, itemstack)) {
```

Comment:
Is the original patch correct here? It seems to undo something intentional from an earlier patch.

Reply:
Patch 0072 added:
` && !com.google.common.base.Objects.equal(defaultReplacement, itemstack`
and patch 325 reverts it.